lib/ansible/modules/network/aos/aos_ip_pool.py

```diff
@@ -0,0 +1,355 @@
+    required: false
+'''
+
+EXAMPLES = '''
+
+- name: "Create an IP Pool with one subnet"
+  aos_ip_pool:
+    session: "{{ session_ok }}"
+    name: "my-ip-pool"
+    subnets: [ 172.10.0.0/16 ]
+    state: present
+
+- name: "Create an IP Pool with multiple subnets"
+  aos_ip_pool:
+    session: "{{ session_ok }}"
+    name: "my-other-ip-pool"
+    subnets: [ 172.10.0.0/16, 192.168.0.0./24 ]
+    state: present
+
+- name: "Check if an IP Pool exist with same subnets by ID"
+  aos_ip_pool:
+    session: "{{ session_ok }}"
+    name: "45ab26fc-c2ed-4307-b330-0870488fa13e"
+    subnets: [ 172.10.0.0/16, 192.168.0.0./24 ]
+    state: present
+
+- name: "Delete an IP Pool by name"
+  aos_ip_pool:
+    session: "{{ session }}"
+    name: "my-ip-pool"
+    state: absent
+
+- name: "Delete an IP pool by id"
+  aos_ip_pool:
+    session: "{{ session }}"
+    id: "45ab26fc-c2ed-4307-b330-0870488fa13e"
+    state: absent
+
+# Save an IP Pool to a file
+
+- name: "Access IP Pool 1/3"
+  aos_ip_pool:
+    session: "{{ session_ok }}"
+    name: "my-ip-pool"
+    subnets: [ 172.10.0.0/16, 172.12.0.0/16 ]
+    state: present
+  register: ip_pool
+
+- name: "Save Ip Pool into a file in JSON 2/3"
+  copy:
+    content: "{{ ip_pool.value | to_nice_json }}"
+    dest: ip_pool_saved.json
+
+- name: "Save Ip Pool into a file in YAML 3/3"
+  copy:
+    content: "{{ ip_pool.value | to_nice_yaml }}"
+    dest: ip_pool_saved.yaml
+
+- name: "Load IP Pool from a JSON file"
+  aos_ip_pool:
+    session: "{{ session_ok }}"
+    content: "{{ lookup('file', 'resources/ip_pool_saved.json') }}"
+    state: present
+
+- name: "Load IP Pool from a YAML file"
+  aos_ip_pool:
+    session: "{{ session_ok }}"
+    content: "{{ lookup('file', 'resources/ip_pool_saved.yaml') }}"
+    content_format: yaml
+    state: present
+
+- name: "Load IP Pool from a Variable"
+  aos_ip_pool:
+    session: "{{ session_ok }}"
+    content:
+      display_name: my-ip-pool
+      id: 4276738d-6f86-4034-9656-4bff94a34ea7
+      subnets:
+        - network: 172.10.0.0/16
+        - network: 172.12.0.0/16
+    content_format: var
+    state: present
+'''
+
+RETURNS = '''
+name:
+  description: Name of the IP Pool
+  returned: always
+  type: str
+  sample: Server-IpAddrs
+
+id:
+  description: AOS unique ID assigned to the IP Pool
+  returned: always
+  type: str
+  sample: fcc4ac1c-e249-4fe7-b458-2138bfb44c06
+
+value:
+  description: Value of the object as returned by the AOS Server
+  returned: always
+  type: dict
+  sample: {'...'}
+'''
+
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.aos import get_aos_session, find_collection_item, do_load_resource, check_aos_version, content_to_dict
+
+def get_list_of_subnets(ip_pool):
+    subnets = []
+
+    for subnet in ip_pool.value['subnets']:
+        subnets.append(subnet['network'])
+
+    return subnets
+
+def create_new_ip_pool(ip_pool, name, subnets=[]):
```

Comment:
lib/ansible/modules/network/aos/aos_ip_pool.py:191:1: M511 - mutable default arg of type List

There is only one caller of create_new_ip_pool() and it provides all three args, so subnets doesnt need to be keyword arg or optional.